src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs

```diff
@@ -79,14 +80,40 @@ public ServiceManagerBuilder WithNewtonsoftJson(Action<NewtonsoftServiceHubProto
         }
 
         /// <summary>
-        /// Uses Newtonsoft.Json library to serialize messages sent to SignalR.
+        /// Uses Newtonsoft.Json library to serialize messages sent to SignalR clients.
         /// </summary>
         /// <returns>The <see cref="ServiceHubContextBuilder"/> instance itself.</returns>
         public ServiceManagerBuilder WithNewtonsoftJson()
         {
             return WithNewtonsoftJson(o => { });
         }
 
+        /// <summary>
+        /// Sets the SignalR hub protocols to serialize messages sent to SignalR clients.
+        /// </summary>
+        /// <remarks><para>Currently it only works for <b>persistent</b> mode. The support for <b>transiet(default)</b> mode is to be done. </para> 
+        /// <para>Calling this method clears the default hub protocol.</para></remarks>
+        /// <param name="hubProtocols">Only the protocols named "json" or "messagepack" are allowed.</param>
+        /// <returns>The <see cref="ServiceHubContextBuilder"/> instance itself.</returns>
+        public ServiceManagerBuilder WithHubProtocols(IEnumerable<IHubProtocol> hubProtocols)
+        {
+            if (hubProtocols == null)
+            {
+                throw new ArgumentNullException(nameof(hubProtocols));
+            }
+            // Allows the user to use MessagePack only.
+            _services.RemoveAll<IHubProtocol>();
+            foreach (var hubProtocol in hubProtocols)
+            {
+                if (!hubProtocol.Name.Equals(Constants.Protocol.Json, StringComparison.OrdinalIgnoreCase) && !hubProtocol.Name.Equals(Constants.Protocol.MessagePack, StringComparison.OrdinalIgnoreCase))
```

Comment:
is the protocol name case-insensitive?

Reply:
Yes, consistent with SignalR implementation https://source.dot.net/#Microsoft.AspNetCore.SignalR.Core/Internal/DefaultHubProtocolResolver.cs,22